### PR TITLE
ci: gate auto-merge on Copilot review for staging PRs

### PR DESCRIPTION
**Pull request by eddimull**
## What

Makes the `auto-merge` job in `main.yml` base-branch-aware instead of merging any passing PR with no review.

| PR base | Gate to auto-merge |
|---|---|
| `staging` | all tests pass **AND** a Copilot review has been posted |
| `master` | all tests pass only |
| `dev` | all tests pass only |

## How

- `auto-merge` now also triggers on `pull_request_review`, so it re-evaluates when Copilot's automatic review lands — in either order relative to the tests finishing.
- On each trigger it re-checks the real state via the API: every check-run on the PR head SHA must be green, and for `staging` it additionally requires ≥1 Copilot review (any verdict). If a gate isn't met it exits cleanly and a later event re-evaluates.
- Copilot reviews every PR via the repo's **automatic code-review** rule, so nothing in CI requests it.

## Notes

- The Copilot-review check matches any reviewer login containing `copilot` (case-insensitive), so it's robust to the exact bot login.
- Policy enforcement against *manual* merges still depends on branch protection requiring the relevant status checks — out of scope for this PR.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -2,32 +2,103 @@ name: CI Pipeline
 
 on:
   pull_request:
-    types: [opened, synchronize]
+    types: [opened, synchronize, reopened]
     branches: [master, staging, dev]
+  # Re-evaluate auto-merge when a review lands (e.g. Copilot finishes its review
+  # on a staging PR after the tests have already passed).
+  pull_request_review:
+    types: [submitted]
```
[the remark this answers is not in the document]
Comment (eddimull) [reply .github/workflows/main.yml:10]: Fixed in 817512d. Added an explicit base allowlist (`master|staging|dev`) at the top of the merge step; any other base (e.g. `release/*`) now exits without merging instead of falling through to the no-review-gate default.

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -2,32 +2,103 @@ name: CI Pipeline
...
+          #    job itself is excluded so it doesn't wait on its own "in_progress".
+          echo "Checking combined check status for $HEAD_SHA..."
+          rollup=$(gh api "repos/$REPO/commits/$HEAD_SHA/check-runs" --paginate \
+            --jq '[.check_runs[] | select(.name != "auto-merge")] as $runs
+                  | { incomplete: ([ $runs[] | select(.status != "completed") ] | length),
+                      failed: ([ $runs[] | select(.conclusion != null and (.conclusion | IN("success","neutral","skipped") | not)) ] | length) }')
+          incomplete=$(echo "$rollup" | jq -s -r 'reduce .[] as $r ({incomplete:0,failed:0}; {incomplete:(.incomplete+$r.incomplete), failed:(.failed+$r.failed)}) | .incomplete')
+          failed=$(echo "$rollup" | jq -s -r 'reduce .[] as $r ({incomplete:0,failed:0}; {incomplete:(.incomplete+$r.incomplete), failed:(.failed+$r.failed)}) | .failed')
+          echo "  incomplete=$incomplete failed=$failed"
+
```
[the remark this answers is not in the document]
Comment (eddimull) [reply .github/workflows/main.yml:99]: Addressed in 817512d. In this repo the `pull_request` check-runs do attach to the head SHA (verified on this PR — frontend/backend checks are present on the head commit), so the head-SHA query is correct here. But to close the "missing check-runs read as all-green" gap regardless, the gate now also requires the expected test contexts (`laravel-tests|backend|frontend`) to be present AND completed-successfully on the SHA before merging; an empty/partial result blocks the merge and a later event re-evaluates.
